src/modules/transformers/displace.rs

```diff
@@ -0,0 +1,138 @@
+// Copyright 2016 The Noise-rs Developers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use num_traits::Float;
+use math::{Point2, Point3, Point4};
+use NoiseModule;
+
+/// Noise Module that uses multiple source modules to displace each coordinate
+/// of the input value before returning the output value from the `source` module.
+pub struct Displace<Source, XDisplace, YDisplace, ZDisplace, UDisplace> {
```

Comment:
What if it's 3d noise, and you only want to mutate the x and z components? `(XDisplace, ZDisplace)` would just be treated as `(XDisplace, YDisplace)`, otherwise you need to specify a `Constant::new(0)` for the `YDisplace` in `(XDisplace, YDisplace, ZDisplace)`. Extend this argument, and you might as well just define all dimensions on initialization.

Ideally, I'd have `Option<T>` for each displacer, and use `.set_x_displace(T).set_y_displace(U)` etc., but the compiler requires that all types be defined on struct initialization. I should add a note to the documentation for this that any dimension not being displaced needs to have `Constant::new(0)` defined as the module.